File: backend/Elearning.API/Models/QuizQuestion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Elearning.API.Models;

[Table("QuizQuestion")]
[Index("QuizId", Name = "IX_QuizQuestion_QuizId")]
public partial class QuizQuestion
{
    [Key]
    public int QuizQuestionId { get; set; }

    public int QuizId { get; set; }

    public string QuestionText { get; set; } = null!;

    [StringLength(50)]
    public string QuestionType { get; set; } = null!;

    [Column(TypeName = "decimal(5, 2)")]
    public decimal Points { get; set; }

    public bool IsActive { get; set; }

    public int OrderIndex { get; set; }

    [ForeignKey("QuizId")]
    [InverseProperty("QuizQuestions")]
    public virtual Quiz Quiz { get; set; } = null!;

    [InverseProperty("QuizQuestion")]
    public virtual ICollection<QuizAnswerOption> QuizAnswerOptions { get; set; } = new List<QuizAnswerOption>();

    [InverseProperty("QuizQuestion")]
    public virtual ICollection<QuizAttemptAnswer> QuizAttemptAnswers { get; set; } = new List<QuizAttemptAnswer>();
}

File: frontend/elearning-web/src/types/quizQuestion.ts
export interface QuizQuestionDto {
  id: number;
  quizId: number;
  quizTitle?: string | null;
  questionText: string;
  questionType: string;
  points: number;
  orderIndex: number;
  isActive: boolean;
}

export interface QuizQuestionCreateDto {
  quizId: number;
  questionText: string;
  questionType: string;
  points: number;
  orderIndex: number;
}

export interface QuizQuestionEditDto {
  id: number;
  quizId: number;
  questionText: string;
  questionType: string;
  points: number;
  orderIndex: number;
}
